pennylane_lightning/core/src/simulators/lightning_qubit/gates/DynamicDispatcher.hpp

```diff
@@ -122,6 +119,16 @@ template <typename PrecisionT> class DynamicDispatcher {
                        PairHash>
         matrix_kernels_{};
 
+    std::unordered_map<KernelType, std::string> kernel_names_{};
+
+    // Defining controlled gates before `kernel_names_` triggers
+    // `assignKernelForOp` before registering kernels at run time.
```

Comment:
Good find on this. I'd have anticipated clang-format to have notified us on this, but I guess since it is runtime that may  have been hard to reason.

Also, we can avoid keeping these comments here, as they'll be attached to the commit message and PR, so feel free to remove them.